python/hail/expr/functions.py

```diff
@@ -215,6 +301,15 @@ def ctt(c1, c2, c3, c4, min_cell_count):
 def Dict(keys, values):
     """Creates a dictionary from a list of keys and a list of values.
 
+    Examples
+    --------
+    .. doctest::
+
+        >>> eval_expr(f.Dict(['foo', 'bar', 'baz'], [1, 2, 3]))
```

Comment:
This is a bit rough, eh? Can we do `f.Dict({'bar': 2, 'baz': 3, 'foo': 1})`?

Reply:
The only time this is necessary is for constructing a dict from existing arrays. You wouldn't need to use the dict constructor for what you wrote -- that's naturally interpretable as a dict expression! 😃